README.md

```diff
@@ -549,7 +549,7 @@ class ProjectPolicy < ApplicationPolicy
   def create?
     if user.has_paid_subscription?
       if user.project_limit_reached?
-        raise Pundit::NotAuthorizedError, reason: 'user.project_limit_reached'
+        raise Pundit::NotAuthorizedError, reason: 'user.project_limit_reached', record: user, query: :create
```

Comment:
Do we need this `record`/`query`? It seems like they're not used - just reason.

Reply:
I suppose yes, because in this issue https://github.com/varvet/pundit/issues/637 is what exactly happened when no corresponding reason is found in `I18n` then use sees default message, and to show it correctly Pundit is required record and query else it will look like `not allowed to  this NilClass`

Reply:
Sure, but why would we want to display a default message in this case?

If there's no corresponding reason in I18n, then in my opinion it should fail: If I set a `reason`, then I'm saying "I want a specific message to be displayed - not the default message", so if it's not possible to display a specific message (because the user forgot to add a key in the locales) then the behaviour should be to fail, not to cover up the fact that it failed.

I'm open to the idea that there's some valid use-case where a reason is specified, but the corresponding locale text isn't specified, but I can't think of one: can you?

I think the controller logic you had before was reasonable:

"If there's a reason set on the error, then look it up in I18n, otherwise show the `message` from the error"

Reply:
It's reasonable, I agree